### ISSUE TITLE
Add generic NVME label to Instances

### ISSUE THREAD

**Issue by sidewinder12s** [aws/karpenter#5238]
### Description

**What problem are you trying to solve?**

Make scheduling to instances that support instance store NVME easier.

There currently is not a generic way to request that a pod be scheduled on a `d` type instance with instance store NVME drives available. While it'd still be up to a node template to mount them, it'd be nice if we had a generic label for it so that we're able to select that & request ephemeral-storage in our pod requests so that Karpenter can make a more informed decision. Currently we'd have to pick an instance and request `karpenter.k8s.aws/instance-local-nvme` and a specific NVME amount which is not flexible. 

I see in Slack that this was one recommendation, though I wonder how that might interact with ephemeral-storage requests.

```yaml
key: karpenter.k8s.aws/instance-local-nvme
operator: Gt
values: ["100"]
```

**How important is this feature to you?**

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by ellistarn** [comment]
```
key: karpenter.k8s.aws/instance-local-nvme
operator: Exists
```
🪄

**Comment by sidewinder12s** [comment]
Oh, that's even better. I keep forgetting that Kubernetes labels are actually fairly expressive. 

Is there a spot in the Karpenter docs yet where you're collecting these operational patterns?

**Comment by ellistarn** [comment]
Exists is definitely a lesser known feature. Would you be interested in contributing these docs improvements?

**Comment by sidewinder12s** [comment]
Yep, I was wondering if you had a spot in the docs yet for more of these operational patterns folks might find useful.

**Comment by ellistarn** [comment]
> Does Karpenter take into account ephemeral-storage requests during scheduling?

https://github.com/aws/karpenter-provider-aws/issues/2394. Highly upvoted feature

> Yep, I was wondering if you had a spot in the docs yet for more of these operational patterns folks might find useful,

Can you add something here https://karpenter.sh/docs/concepts/scheduling/#advanced-scheduling-techniques
